src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -334,9 +345,27 @@ private boolean buildAndTag() throws MacroEvaluationException, IOException, Inte
         private boolean dockerPushCommand() throws InterruptedException, MacroEvaluationException, IOException {
         	List<String> result = new ArrayList<String>();
         	for (String tag: getNameAndTag()) {
+
         		result.add("docker push " + tag);
         	}
-        	return executeCmd(result);
+            boolean lastResult = executeCmd(result);
+            int retryNo=parse(getRetryBuild());
+        	while (!lastResult && retryNo>0)
+            {
+                logger.log(Level.INFO, "Retrying push command : {0} attempt", retryNo);
+                lastResult = executeCmd(result);
+                retryNo--;
+            }
+            return lastResult;
+        }
+
+        private  int parse(String p) {
```

Comment:
you can use NumberUtils.toInt, nos need for a new parse method
 https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/math/NumberUtils.html#toInt(java.lang.String)